src/setfit/trainer.py

```diff
@@ -372,22 +373,37 @@ def train(
                 show_progress_bar=True,
             )
 
-    def evaluate(self):
-        """Computes the metrics for a given classifier."""
+    def evaluate(self, **evaluate_kwargs):
+        """
+        Computes the metrics for a given classifier.
+
+        Args:
+            **evaluate_kwargs: Keyword arguments to be passed to the `evaluate` method of the classifier.
+        """
+
         self._validate_column_mapping(self.eval_dataset)
         eval_dataset = self.eval_dataset
+
         if self.column_mapping is not None:
             logger.info("Applying column mapping to evaluation dataset")
             eval_dataset = self._apply_column_mapping(self.eval_dataset, self.column_mapping)
-        metric_config = "multilabel" if self.model.multi_target_strategy is not None else None
-        metric_fn = evaluate.load(self.metric, config_name=metric_config)
+
         x_test = eval_dataset["text"]
         y_test = eval_dataset["label"]
 
         logger.info("***** Running evaluation *****")
         y_pred = self.model.predict(x_test)
 
-        return metric_fn.compute(predictions=y_pred, references=y_test)
+        if isinstance(self.metric, str):
+            metric_config = "multilabel" if self.model.multi_target_strategy is not None else None
+            metric_fn = evaluate.load(self.metric, config_name=metric_config)
+
+            return metric_fn.compute(predictions=y_pred, references=y_test, **evaluate_kwargs)
+
+        if callable(self.metric):
```

Comment:
Should this better be an `elif`?

```
        elif callable(self.metric):
```

Reply:
Yes, then we could end the logic with an `else` to raise the error :)

Reply:
Yes. That was also my first idea. But if we do that we could maybe better do it in the `__init__` function - because there
`self.metric` is set.